sources/kafka_pipeline.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Dict
+
+import confluent_kafka  # type: ignore
+import dlt
+
+try:
+    from .kafka import kafka_consumer  # type: ignore
+except ImportError:
+    from kafka import kafka_consumer
+
+
+def load_from_several_topics() -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="kafka_pipeline",
+        destination="postgres",
+        dataset_name="kafka_messages",
+    )
+
+    data = kafka_consumer(["purchases", "books"])
+
+    info = pipeline.run(data)
+    print(info)
+
+
+def load_with_custom_processor() -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="kafka_pipeline",
+        destination="postgres",
+        dataset_name="kafka_messages",
+    )
+
+    def custom_msg_processor(msg: confluent_kafka.Message) -> Dict[str, Any]:
+        """Convert the given message before loading into the destination.
+
+        Args:
+            msg (confluent_kafka.Message): Message read from the cluster.
+
+        Returns:
+            dict: converted message.
+        """
+        return {
+            "_kafka": {
+                "topic": msg.topic(),  # required field
+                "key": msg.key().decode("utf-8"),
+                "partition": msg.partition(),
```

Comment:
there are some relevant information that must be stored:
* msg.offset - helpful for further downstream de-duplications
* msg.timestamp() - when the event was inserted into kafka- again really helpful for further downstream de-duplications
* event.headers() - could be relevant at time